lib/RestructureCFG/InlineDispatcherSwitch.cpp

```diff
@@ -737,13 +737,140 @@ inlineDispatcherSwitchImpl(ASTTree &AST,
   rc_return Node;
 }
 
-ASTNode *inlineDispatcherSwitch(ASTTree &AST, ASTNode *RootNode) {
+static RecursiveCoroutine<ASTNode *>
+simplifySuperfluousNonLocalCFImpl(ASTNode *Node,
+                                  const FallThroughScopeTypeMap
+                                    &FallThroughScopeMap) {
+  switch (Node->getKind()) {
+  case ASTNode::NK_List: {
+    SequenceNode *Seq = llvm::cast<SequenceNode>(Node);
+
+    // In place of a sequence node, we just need to inspect all the nodes in the
+    // sequence
+    bool ShouldSimplify = false;
+    for (ASTNode *&N : Seq->nodes()) {
+      if (not ShouldSimplify) {
+
+        // Compute the `FallThrough information after each `SequencenNode` is
+        // processed
+        N = rc_recur simplifySuperfluousNonLocalCFImpl(N, FallThroughScopeMap);
+        FallThroughScopeType ReturnType = FallThroughScopeType::Return;
```

Comment:
We should do something about calls to noreturn functions too.

Reply:
In rewriting this as the `RemoveDeadCode` beautify pass, I changed the approach here. Basically, I know look for everything that does not `fallthrough`, and this encompasses `noreturn` functions too. Code [here](https://github.com/revng/revng-c/blob/266c681abb4668354bbbbc5cb6802ad4c583ac30/lib/RestructureCFG/RemoveDeadCode.cpp#L36)